### PR TITLE
refactor: icon buttons

### PR DESCRIPTION
**Pull request by joyce-shi**
follow-up from #476. to test, make sure that all icon buttons behave as they previously did

### CODE REVIEW

File: frontend/src/components/admin/question-creation/QuestionElementItem.tsx [hunk 1]
```diff
@@ -152,10 +151,12 @@ const QuestionElementItem = ({
         </Box>
         {renderQuestionContent(content)}
         <IconButton
+          _hover={{ color: "blue.100" }}
+          aria-label="delete-icon-button"
```
Comment (jfdoming) [inline frontend/src/components/admin/question-creation/QuestionElementItem.tsx]: Does this (and similar) read out fine to screen readers with all the hyphens? Normally I'd put spaces instead to make it more readable

File: frontend/src/components/pages/teacher/DisplayClassroomPage/index.tsx [hunk 2]
```diff
@@ -176,14 +176,14 @@ const DisplayClassroomsPage = () => {
...
-            color="blue.300"
+            aria-label="edit-classroom"
             icon={<EditOutlineIcon />}
-            minW={0}
             onClick={() => {
               classroomFormMethods.reset();
               onClassroomModalOpen();
             }}
+            size="icon"
+            variant="icon"
```
Comment (jfdoming) [inline frontend/src/components/pages/teacher/DisplayClassroomPage/index.tsx:186]: What's the purpose of the extra variant here and throughout, out of curiosity? Wondering if we could just style the default variant instead, or give this one a more descriptive name if we truly need more than one

Comment (joyce-shi) [reply frontend/src/components/pages/teacher/DisplayClassroomPage/index.tsx:186]: `<IconButton />` gets its theming from the `button.ts` file (i.e under the hood it's actually a button). So we can't style the default variant.

What kind of descriptive name did you have in mind? I meant for the variant to just store the standard `color` and `_hover` attributes for `icon` buttons.

File: frontend/src/components/admin/question-creation/QuestionElementItem.tsx (hunk 1 above)
Comment (joyce-shi) [reply frontend/src/components/admin/question-creation/QuestionElementItem.tsx]: see latest commit

File: frontend/src/components/pages/teacher/DisplayClassroomPage/index.tsx (hunk 2 above)
Comment (jfdoming) [reply frontend/src/components/pages/teacher/DisplayClassroomPage/index.tsx:186]: Gotcha, it's fine as is then. We could potentially make a wrapper component just to specify the default variant and size, but probably not worth it then.

**Review by jfdoming — APPROVED** [review]
Looks good to me!

**Pull request merged**